src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1360,24 +1360,30 @@ private function registerTranslatorConfiguration(array $config, ContainerBuilder
             return;
         }
 
+        $locales = $config['enabled_locales'] ?? [];
+
         foreach ($config['providers'] as $name => $provider) {
-            if (!$config['enabled_locales'] && !$provider['locales']) {
-                throw new LogicException(sprintf('You must specify one of "framework.translator.enabled_locales" or "framework.translator.providers.%s.locales" in order to use translation providers.', $name));
+            if (!$config['enabled_locales']) {
+                if (!$provider['locales']) {
+                    throw new LogicException(sprintf('You must specify one of "framework.translator.enabled_locales" or "framework.translator.providers.%s.locales" in order to use translation providers.', $name));
+                }
+
+                $locales = array_merge($locales, $provider['locales']);
```

Comment:
Is array_merge really the intended way here? Shouldn't the locale options for the provider be what controls what locales get sent to the provider, even if we have more locales enabled?

Reply:
Actually we want to pass to the commands all locales supported by all configured providers, in order to read local translations. That's why the `locales` property of the `TranslationPushCommand` has `enabledLocales` as default value (https://github.com/symfony/symfony/pull/41504/files#diff-a00f86baf85c337ee61148e5fff8d189edf224ef40382518307359f579f40e6aR66).

Then, the `FilteringProvider` class is a decorator for each actual provider, and in its `read` method, we intersect domains and locales : the ones come from the provider configurations, the others come from the `read` call from the command class.

But I'm ok, we need to review this part, to make it more explicit and more logical. One of the solution might be to use only locales of providers configuration, make this parameter mandatory? WDYT @nicolas-grekas, in terms of design?

Reply:
I've no strong opinion. I'm fine with the current design.
But if someone disagrees, I'd be happy to know why.

Reply:
Are @OskarStark or @Nyholm has an opinion on it?

Reply:
ping @OskarStark @Nyholm